### ISSUE TITLE
mavericks segfault, cmake/tools for minuit2 & root (Trac #868)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#862]
The mavericks buildbot is failing with a segfault in the Minuit2 unit tests in lilliput. I tried to reproduce this on valefar, but both Debug and Release builds run the lilliput tests without any problem.

Then I noticed that apparently I made some changes to cmake/ quite some time ago already (see patch below):
* standalone minuit2 should be found even if it is not version 5.24
* some users may have root 5.34.19 installed
* (unrelated) tooldef message for unfound tools is weird

Later I will verify that with the svn trunk version of cmake I will indeed reproduce the crash, but I think Meike already did that, so there is not much doubt about that.

I am not sure if my patch is indeed the correct solution, but it at least suggests that the cause of the crash is likely related to how cmake currently finds minuit2 and root.

```text
boersma@valefar:/build/boersma/icerec/trunk/src$ svn diff cmake
Index: cmake/tools/minuit2.cmake
===================================================================
--- cmake/tools/minuit2.cmake	(revision 2063)
+++ cmake/tools/minuit2.cmake	(working copy)
@@ -17,10 +17,19 @@
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <http://www.gnu.org/licenses/>
 #
+#TOOLDEF (minuit2
+#    include/Minuit2-5.24.00
+#    Minuit2/MnConfig.h
+#    lib/Minuit2-5.24.00
+#    NONE  # bin is n/a, placeholder
+#    Minuit2
+#    )
+
+#macro(tooldef tool_ incdir incfile libdir bindir)
 TOOLDEF (minuit2
-    include/Minuit2-5.24.00
+    include/Minuit2
     Minuit2/MnConfig.h
-    lib/Minuit2-5.24.00
+    lib/Minuit2
     NONE  # bin is n/a, placeholder
     Minuit2
     )
Index: cmake/tools/root.cmake
===================================================================
--- cmake/tools/root.cmake	(revision 2063)
+++ cmake/tools/root.cmake	(working copy)
@@ -28,7 +28,7 @@
 
 if (USE_ROOT)
   if(NOT ROOT_VERSION)
-    foreach(ROOTVER 5.34.18 5.34.04 5.34.03 5.32.00 5.30.06 5.30.05 5.30.00 5.28.00h 5.28.00d 5.28.00 5.27.06b 5.26.00e 5.24.00b 5.24.00 5.20.00 5.18.00)
+    foreach(ROOTVER 5.34.19 5.34.18 5.34.04 5.34.03 5.32.00 5.30.06 5.30.05 5.30.00 5.28.00h 5.28.00d 5.28.00 5.27.06b 5.26.00e 5.24.00b 5.24.00 5.20.00 5.18.00)
       if(IS_DIRECTORY ${I3_PORTS}/root-v${ROOTVER})
         set(ROOT_VERSION "${ROOTVER}")
         set(ROOT_${ROOT_VERSION}_LIBS Core Cint RIO Net Hist Graf Graf3d Gpad Tree Rint Postscript Matrix Physics MathCore Thread Minuit Gui)
Index: cmake/tools/tooldef.cmake
===================================================================
--- cmake/tools/tooldef.cmake	(revision 2063)
+++ cmake/tools/tooldef.cmake	(working copy)
@@ -103,8 +103,10 @@
 
   if (NOT ${TOOL}_CONFIG_ERROR)
     set(${TOOL}_FOUND TRUE CACHE BOOL "Tool '${TOOL}' found successfully" FORCE)
+    colormsg(HICYAN "${tool_} WAS FOUND")
   else (NOT ${TOOL}_CONFIG_ERROR)
-    set(${TOOL}_FOUND FALSE CACHE BOOL "Tool '${TOOL}' found successfully" FORCE)
+    set(${TOOL}_FOUND FALSE CACHE BOOL "Tool '${TOOL}' NOT found successfully...!" FORCE)
+    colormsg(HICYAN "${tool_} WAS NOT FOUND OR HAD PROBLEMS")
   endif (NOT ${TOOL}_CONFIG_ERROR)
 
 endmacro(tooldef)
```

Migrated from https://code.icecube.wisc.edu/ticket/868
```json
{
    "status": "closed", 
    "changetime": "2019-01-11T23:12:33", 
    "description": "The mavericks buildbot is failing with a segfault in the Minuit2 unit tests in lilliput. I tried to reproduce this on valefar, but both Debug and Release builds run the lilliput tests without any problem.\n\nThen I noticed that apparently I made some changes to cmake/ quite some time ago already (see patch below):\n* standalone minuit2 should be found even if it is not version 5.24\n* some users may have root 5.34.19 installed\n* (unrelated) tooldef message for unfound tools is weird\n\nLater I will verify that with the svn trunk version of cmake I will indeed reproduce the crash, but I think Meike already did that, so there is not much doubt about that.\n\nI am not sure if my patch is indeed the correct solution, but it at least suggests that the cause of the crash is likely related to how cmake currently finds minuit2 and root.\n\n{{{\nboersma@valefar:/build/boersma/icerec/trunk/src$ svn diff cmake\nIndex: cmake/tools/minuit2.cmake\n===================================================================\n--- cmake/tools/minuit2.cmake\t(revision 2063)\n+++ cmake/tools/minuit2.cmake\t(working copy)\n@@ -17,10 +17,19 @@\n #  You should have received a copy of the GNU General Public License\n #  along with this program.  If not, see <http://www.gnu.org/licenses/>\n #\n+#TOOLDEF (minuit2\n+#    include/Minuit2-5.24.00\n+#    Minuit2/MnConfig.h\n+#    lib/Minuit2-5.24.00\n+#    NONE  # bin is n/a, placeholder\n+#    Minuit2\n+#    )\n+\n+#macro(tooldef tool_ incdir incfile libdir bindir)\n TOOLDEF (minuit2\n-    include/Minuit2-5.24.00\n+    include/Minuit2\n     Minuit2/MnConfig.h\n-    lib/Minuit2-5.24.00\n+    lib/Minuit2\n     NONE  # bin is n/a, placeholder\n     Minuit2\n     )\nIndex: cmake/tools/root.cmake\n===================================================================\n--- cmake/tools/root.cmake\t(revision 2063)\n+++ cmake/tools/root.cmake\t(working copy)\n@@ -28,7 +28,7 @@\n \n if (USE_ROOT)\n   if(NOT ROOT_VERSION)\n-    foreach(ROOTVER 5.34.18 5.34.04 5.34.03 5.32.00 5.30.06 5.30.05 5.30.00 5.28.00h 5.28.00d 5.28.00 5.27.06b 5.26.00e 5.24.00b 5.24.00 5.20.00 5.18.00)\n+    foreach(ROOTVER 5.34.19 5.34.18 5.34.04 5.34.03 5.32.00 5.30.06 5.30.05 5.30.00 5.28.00h 5.28.00d 5.28.00 5.27.06b 5.26.00e 5.24.00b 5.24.00 5.20.00 5.18.00)\n       if(IS_DIRECTORY ${I3_PORTS}/root-v${ROOTVER})\n         set(ROOT_VERSION \"${ROOTVER}\")\n         set(ROOT_${ROOT_VERSION}_LIBS Core Cint RIO Net Hist Graf Graf3d Gpad Tree Rint Postscript Matrix Physics MathCore Thread Minuit Gui)\nIndex: cmake/tools/tooldef.cmake\n===================================================================\n--- cmake/tools/tooldef.cmake\t(revision 2063)\n+++ cmake/tools/tooldef.cmake\t(working copy)\n@@ -103,8 +103,10 @@\n \n   if (NOT ${TOOL}_CONFIG_ERROR)\n     set(${TOOL}_FOUND TRUE CACHE BOOL \"Tool '${TOOL}' found successfully\" FORCE)\n+    colormsg(HICYAN \"${tool_} WAS FOUND\")\n   else (NOT ${TOOL}_CONFIG_ERROR)\n-    set(${TOOL}_FOUND FALSE CACHE BOOL \"Tool '${TOOL}' found successfully\" FORCE)\n+    set(${TOOL}_FOUND FALSE CACHE BOOL \"Tool '${TOOL}' NOT found successfully...!\" FORCE)\n+    colormsg(HICYAN \"${tool_} WAS NOT FOUND OR HAD PROBLEMS\")\n   endif (NOT ${TOOL}_CONFIG_ERROR)\n \n endmacro(tooldef)\n}}}\n", 
    "reporter": "boersma", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547248353602202", 
    "component": "cmake", 
    "summary": "mavericks segfault, cmake/tools for minuit2 & root", 
    "priority": "normal", 
    "keywords": "mavericks, minuit2, root, cmake", 
    "time": "2015-02-11T18:14:52", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150211T21:14:15: 
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "nega"

**Comment by icecube-trac** [comment]
Trac update at 20150212T00:25:10: 
* nega commented:

> In changeset 2064:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2064"
> minuit/root detection update from boersma
> 
> closes #862
> ```


* nega changed status from "assigned" to "closed"
* nega changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190111T23:12:33: nwhitehorn commented:

> In changeset 169925:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="169925"
> minuit/root detection update from boersma
> 
> closes #862
> ```
